lib/ui/painting/display_list_deferred_image_gpu.h

```diff
@@ -83,11 +103,15 @@ class DlDeferredImageGPU final : public DlImage {
 
    private:
     const SkImageInfo image_info_;
-    sk_sp<DisplayList> display_list_;
+    std::variant<sk_sp<DisplayList>, std::shared_ptr<LayerTree>>
+        display_list_or_layer_tree_;
```

Comment:
Why not just keep this as a `sk_sp<DisplayList>`?

This makes the API a bit more confusing to read when we're just going to flatten it to a displaylist the first time we have to use it anyway.

Reply:
I see now. This makes it easier to return from the UI thread in the scene case.

I think we should just make `SnapshotDisplayList` take a `std::shared_ptr<LayerTree>` param. If that param is not null, it has to be flattened and set to the dispaly list, and that's only called as non-null from `MakeFromLayerTree`.